os/arch/arm/src/armv7-r/arm_assert.c

```diff
@@ -238,39 +240,47 @@ void get_symbol(unsigned long search_addr, char *buffer)
 	while (first <= last) {
 		fseek(pFile, mid, SEEK_SET);
 
-		// Extract characters from file and store in character c
+		/* If the file pointer is in the mid of the line, make sure
+		 * it's been properly moved to start of next line
+		 */
 		for (c = getc(pFile); c != '\n'; c = getc(pFile)) {
+			if (c == EOF) {
+				lldbg("Reached end of file and couldn't find symbol\n");
+				fclose(pFile);
+				return -1;
+			}
 		}
-		n = 0;
+
+		word = 0;
 		// Read 2 lines and Split the string as words
-		for (k = 0; k < 2; k++) {
+		for (read_line = 0; read_line < 2; read_line++) {
 			fgets(line, 128, pFile);
-			i = 0;
-			j = 0;
-			while (line[i] != '\0') {
-				if (line[i] != ' ') {
-					data[n][j++] = line[i];
+			ch_in_line = 0;
+			ch_in_word = 0;
+			while (line[ch_in_line] != '\0') {
+				if (line[ch_in_line] != ' ') {
+					data[word][ch_in_word++] = line[ch_in_line];
 				} else {
-					data[n][j++] = '\0';//insert NULL
-					n++;
-					j = 0;
+					data[word][ch_in_word++] = '\0';
```

Comment:
As before I commented, data[word][ch_in_word] = '\0'; is enough.
Do not you agree with this?

Reply:
somehow missed it :) . It's done